chef/cookbooks/barclamp/libraries/nic.rb

```diff
@@ -743,6 +744,20 @@ def slaves
       ports.map{ |i| ::Nic.new(i) }
     end
 
+    def add_slave(slave)
+      slave = self.class.coerce(slave)
+      unless ::Nic.exists?(slave)
+        raise ::ArgumentError.new("#{slave} does not exist, cannot add to bridge#{@nic}")
+      end
+      return if slaves.member?(slave)
+      if current_master = slave.master
+        current_master.remove_slave(slave)
+      end
+      slave.up
+      usurp(slave)
+      ::Kernel.system("ovs-vsctl add-port #{@nic} #{slave}")
```

Comment:
It's worth mentioning that the Bond class actually returns the slave from this method (also in the `return if` bit a few lines above).